### PR TITLE
he: Handle migration to hosts that use systemd-coredump

### PR DESCRIPTION
**Pull request by tinez**
This PR adds proper handling of a situation where the HE is migrated
from a host that was using abrt to a host that's using systemd-coredump.

Bug-Url: https://bugzilla.redhat.com/2101481
Signed-off-by: Marcin Sobczyk <msobczyk@redhat.com>

### CODE REVIEW

**Review by arachmani — APPROVED** [review]
LGTM

File: roles/hosted_engine_setup/tasks/initial_clean.yml [hunk 1]
```diff
@@ -56,12 +60,22 @@
         src: /usr/share/abrt/conf.d/plugins/vmcore.conf,
         dest: /etc/abrt/plugins/vmcore.conf
       }
-    when: abrt_exists.stat.exists
+    when: abrt_vdsm_config.stat.exists and abrt_installed_config.stat.exists
```
Comment (mwperina) [inline roles/hosted_engine_setup/tasks/initial_clean.yml:63]: Please use "more readable" :-) syntax:
```
    when:
      - abrt_vdsm_config.stat.exists
      - abrt_installed_config.stat.exists
```

File: roles/hosted_engine_setup/tasks/initial_clean.yml [hunk 2]
```diff
@@ -56,12 +60,22 @@
...
+  - name: Remove vdsm's abrt config files
+    file:
+      state: absent
+      path: "{{ item }}"
+    with_items:
+      - /etc/abrt/abrt-action-save-package-data.conf
+      - /etc/abrt/abrt.conf
+      - /etc/abrt/plugins/CCpp.conf
+      - /etc/abrt/plugins/vmcore.conf
+    when: abrt_vdsm_config.stat.exists and not abrt_installed_config.stat.exists
```
Comment (mwperina) [inline roles/hosted_engine_setup/tasks/initial_clean.yml:78]: Wouldn't it be better to encapsulate `Restore initial abrt config files`, `Restart abrtd service`  and `Remove vdsm's abrt config files` into a `block` and use the condition for the whole block?

**Review by mwperina — CHANGES_REQUESTED** [review]

File: roles/hosted_engine_setup/tasks/initial_clean.yml (hunk 2 above)
Comment (tinez) [reply roles/hosted_engine_setup/tasks/initial_clean.yml:78]: `Remove vdsm's...` uses a different condition - notice the `not`, so the block would include only the first two. Do you think it's still worth it?

Comment (mwperina) [reply roles/hosted_engine_setup/tasks/initial_clean.yml:78]: Ahh sorry, missed that. So the I'd say stay as is, just please reformat the conditions

File: roles/hosted_engine_setup/tasks/initial_clean.yml (hunk 1 above)
Comment (tinez) [reply roles/hosted_engine_setup/tasks/initial_clean.yml]: Done, hope the one with `not` will work too

**Review by mwperina — APPROVED** [review]
+1

**Pull request merged**